### ISSUE TITLE
bug in assocTestSingle if there are monomorphic SNPs

### ISSUE THREAD

**Issue by jonaszierer** [UW-GAC/GENESIS#30]
In `assocTestSingle` monomorphic SNPs are filtered in this block (around line 101)

```
freq <- .alleleFreq(gdsobj, geno, sample.index=sample.index,
                                      male.diploid=male.diploid)
                  
# filter monomorphic variants
keep <- .filterMonomorphic(geno, count=n.obs, freq=freq$freq)
if (!all(keep)) {
  var.info <- var.info[keep,,drop=FALSE]
  geno <- geno[,keep,drop=FALSE]
  n.obs <- n.obs[keep]
  freq <- freq[keep,,drop=FALSE]
}
```

But freq is actually a list with the entries `freq` and `MAC`. So the last line should be 

```
 freq$freq <- freq$freq[keep,,drop=FALSE]
 freq$MAC <- freq$MAC[keep,,drop=FALSE]
```

**Comment by jonaszierer** [comment]
sorry the it actually should just be
```
freq$freq <- freq$freq[keep]
freq$MAC <- freq$MAC[keep]
```

**Comment by smgogarten** [comment]
`freq` is a data.frame, so the matrix-type subsetting should work. Can you show the error you're getting?

**Comment by jonaszierer** [comment]
`.alleleFreq` actually returns a list if sex is missing: https://github.com/UW-GAC/GENESIS/blob/a822d2e87e7bd4b5a323081d7a07b6a9fce5fe97/R/utils.R#L65-L75

**Comment by smgogarten** [comment]
Thanks! Fixed in 2bb43fd
